### PR TITLE
Fix Postgres execute error sql no response (#3129)

### LINKED ISSUE #3129: shardingJDBC(4.0.0-RC2) proxy Postgres  execute sql ,  ERROR SQL no return , CLI  no response
## Bug Report

**For English only**, other languages will not accept.

Before report a bug, make sure you have:

- Searched open and closed [GitHub issues](https://github.com/apache/incubator-shardingsphere/issues).
- Read documentation: [ShardingSphere Doc](https://shardingsphere.apache.org/document/current/en/overview).

Please pay attention on issues you submitted, because we maybe need more details. 
If no response **more than 7 days** and we cannot reproduce it on current information, we will **close it**.

Please answer these questions before submitting your issue. Thanks!

### Example codes for reproduce this issue (such as a github link).

postgres psql connect sharing JDBC proxy  ,exec error sql not return ,cli no response.

### PR DESCRIPTION
**Pull request by avalon566**
Fixes #3129.

Changes proposed in this pull request:
- Add hasError method for QueryCommandExecutor.
- Fix Postgres return error response.
- Return backend error response to frontend.

### CODE REVIEW

File: sharding-proxy/sharding-proxy-frontend/sharding-proxy-frontend-mysql/src/main/java/org/apache/shardingsphere/shardingproxy/frontend/mysql/command/query/binary/execute/MySQLComStmtExecuteExecutor.java [hunk 1]
```diff
@@ -60,6 +60,8 @@
     
     private volatile boolean isQuery;
     
+    private volatile boolean hasError;
```
Comment (tuohai666) [inline sharding-proxy/sharding-proxy-frontend/sharding-proxy-frontend-mysql/src/main/java/org/apache/shardingsphere/shardingproxy/frontend/mysql/command/query/binary/execute/MySQLComStmtExecuteExecutor.java]: I am not sure whether we should add this in MySQL frontend. I suggest PostgreSQL handle exceptions the same way an MySQL in CommandExecutorTask. What do you think?

Comment (avalon566) [reply sharding-proxy/sharding-proxy-frontend/sharding-proxy-frontend-mysql/src/main/java/org/apache/shardingsphere/shardingproxy/frontend/mysql/command/query/binary/execute/MySQLComStmtExecuteExecutor.java]: In PostgreSQL, if return an error response when execute sql, you should not write PostgreSQLCommandCompletePacket, so i add this flag. [diff](https://github.com/apache/incubator-shardingsphere/pull/3829/files#diff-9ed7b59514c0a038841321e90a278ddc)

Comment (tuohai666) [reply sharding-proxy/sharding-proxy-frontend/sharding-proxy-frontend-mysql/src/main/java/org/apache/shardingsphere/shardingproxy/frontend/mysql/command/query/binary/execute/MySQLComStmtExecuteExecutor.java]: Can you go through MySQL's exception mechanism and try to abstract PostgreSQL's the same way?

**Comment by tuohai666** [comment]
@avalon566 Can you show me your test case? Is this pr work only for DQL? How DML handle exceptions now?

**Comment by avalon566** [comment]
@tuohai666 Do you mean this [test case](https://github.com/apache/incubator-shardingsphere/pull/3829/files#diff-2e279dfc80cc12d56e17501c4ee79cbd)? DML Like DQL.

**Pull request closed without merging**